### PR TITLE
Upgrade depencencies for python 3.12 / pylint 3.0+

### LINKED ISSUE #676: Upgrade requirements-detector / remove pylint-plugin-utils as a direct dependency
Based on [this conversation](https://github.com/prospector-dev/prospector/issues/670#issuecomment-2397340662), it seems like `pylint-plugin-utils` could be removed as a direct dependency from `pyproject.toml`

Also, in #640 - see [this comment](https://github.com/prospector-dev/prospector/issues/670#issuecomment-2393350433) - the only compatible version of `requirements-detector` with pylint 3 and the current astroid version will be `>=1.3.1` so this should also be updated in `pyproject.toml`. The current version constraints will pull that on a fresh build, however old virtualenvs won't feel the need to update it when updating prospector, this could lead to issues.

(Maybe at the same time, running `poetry update` to get latest versions of other dependencies in the `poetry.lock` file is a good idea)

### PR DESCRIPTION
**Pull request by Pierre-Sassoulas**
This removes pylint-plugin-utils, upgrade requirements-detector to the only version that can work, and prepare for the next release of pylint-django (that is necessary for python 3.12.) 

Fix #676

Refs https://github.com/pylint-dev/pylint-django/pull/444 https://github.com/pylint-dev/pylint-django/releases/tag/v2.6.0

### CODE REVIEW

File: pyproject.toml
Comment (valeriupredoi) [inline pyproject.toml:1]: Looks good, but it'd be very useful, Prospector functionality permitting, to relax the upper pin on flake8, they are already at 7.1-7.2 👍🍺

**Review by sbrunner — APPROVED** [review]

File: pyproject.toml
Comment (Pierre-Sassoulas) [reply pyproject.toml:1]: Done in [2c01a78](https://github.com/prospector-dev/prospector/pull/678/commits/2c01a78e4d9a99e22b543fbde9eb14be641a20af)

**Comment by sbrunner** [comment]
@Pierre-Sassoulas shouldn't we create now the release 1.12.1?

**Comment by valeriupredoi** [comment]
Yes, if the PyPI package is released, then I'll build and release the conda package today 👍

File: pyproject.toml
Comment (valeriupredoi) [reply pyproject.toml:1]: Excellent many thanks @Pierre-Sassoulas 🍺

**Pull request merged**